app/interactors/convert_tts.rb

```diff
@@ -39,12 +39,29 @@ def convert_slot(slot, parallel_id)
       s.deleted_at = nil
       s.start_time = slot.start_time
       s.end_time = slot.end_time
-      s.weeks = slot.weeks
+      s.weeks = convert_weeks_to_ranges(slot.weeks) if slot.weeks
     end
   end
 
   def valid?(slot)
     slot.day != nil
   end
 
+  def convert_weeks_to_ranges(weeks)
+    weeks.split(',').map do |interval|
+      if interval.length > 2
+        # true interval
+        interval_split = interval.split('-')
+        from = Integer(interval_split[0])
+        to = Integer(interval_split[1])
+      else
+        # one week
+        week_number = Integer(interval)
+        from = week_number
+        to = week_number
+      end
+      Sequel.pg_range((from..to))
+    end
+  end
```

Comment:
This is unnecessary verbose and maybe too tolerant of malformed intervals. I would write it as:

```rb
def convert_weeks_to_ranges(weeks)
  weeks.split(',').map do |str|
    from, to = str.split('-', 2)
    Range.new(Integer(from), Integer(to || from))
  end
end
```